src/test/java/picard/sam/BuildBamIndexTest.java

```diff
@@ -41,6 +42,20 @@ public void testBuildBamIndexOK() throws IOException {
         Assert.assertEquals(FileUtils.readFileToByteArray(OUTPUT_INDEX_FILE),  FileUtils.readFileToByteArray(EXPECTED_BAI_FILE));
     }
 
+    @Test
+    public void testBuildBamIndexNoOutputSpecified() throws IOException {
+        final List<String> args = new ArrayList<>();
+        /* First sort, before indexing */
+        new SortSam().instanceMain(new String[]{
+                "I=" + INPUT_FILE,
+                "O=" + OUTPUT_SORTED_FILE,
```

Comment:
The test is called "NoOutputSpecified", but it specifies an output .

Reply:
The existing tests in this file are already a bit sketchy, since they write their outputs into the testdata directory. Its not clear if they're actually overwriting test data or not (I don't think they are, but it LOOKS like they are).